scripts/windows/efcore/add_migration.ps1

```diff
@@ -8,6 +8,15 @@ $environment="dbmigrations-" + $provider.ToLower()
 $repoRoot = git rev-parse --show-toplevel
 $dbContextName = "${moduleName}DbContext"
 $startupProject = "$repoRoot\ConsumerApi"
+$adminUiProject = "$repoRoot\AdminUi\src\AdminUi"
+
+function CompileModelsAdminUi {
+    $cmdAdminUiDbContext = "dotnet ef dbcontext optimize --project $adminUiProject --context AdminUiDbContext --output-dir $repoRoot\AdminUi\src\AdminUi.Infrastructure\CompiledModels --namespace AdminUi.Infrastructure.CompiledModels"
+    $cmdDevicesDbContext = "dotnet ef dbcontext optimize --project $adminUiProject --context DevicesDbContext --output-dir $repoRoot\Modules\Devices\src\Devices.Infrastructure\CompiledModels --namespace Devices.Infrastructure.CompiledModels"
+    Write-Host "Compiling models for '$adminUiProject'..."
```

Comment:
Pleaaaase, stop wasting my time with such **unnecessary** mistakes...! The log message says that you're compiling models for the Admin UI project. But then you compile them for both Admin UI AND the Devices module... Same for the function name.

However. The change was unnecessary anyway. I don't understand why you only compile those two DbContexts now. We need this for all DbContexts. And please don't start copying the command X times. Instead, do it only for the db context for which there is currently a migration being generated.

Reply:
From what I understood, we should only need to run the optimize command on the project we are having issues (in this case the adminUi project). In a normal situation, we would only need to run the following command, without specifying the DbContext:

`dotnet ef dbcontext optimize --project $adminUiProject --output-dir $repoRoot\AdminUi\src\AdminUi.Infrastructure\CompiledModels --namespace AdminUi.Infrastructure.CompiledModels`

But after running the above command, dotnet will say that there are currently two DbContexts in use for the adminUi project, and that we should specify which one we want to optimize. The two DbContexts being the AdminUiDbContext and the DevicesDbContext. The reason for this is that we use the DevicesDbContext for OpenIddict in the adminUi project.

Seeing as the adminUi project does not use the other DbContexts, I'm not sure if we could even run the optimize command for them in the adminUi project, and if it would even make a diference in the performance.

As for the naming, we are only compiling the DbContexts used by the adminUi project so maybe we can change the function name to 'CompileModelsUsedByAdminUi' and the log message to 'Compiling models used by $adminUiProject...'.

Reply:
The problem isn't just with the Admin UI. It's similar for the Consumer API - although less extreme there. But the first request takes about 10-20 times as long as subsequent requests.

So having compiled models for everything seems like it was worth a try.